### PR TITLE
Feature/NAKIJIN-237. Add endpoints to get locality candidates and synonyms

### PR DESCRIPTION
**Pull request by oleg-n-k**
Add endpoints to get locality candidates and synonyms

### CODE REVIEW

File: HappyTravel.Nakijin.Api/Services/AdministratorServices/LocalityCandidatesManagementService.cs [hunk 1]
```diff
@@ -124,6 +125,16 @@ Task AddSynonyms(List<LocalityCandidate> localityCandidates)
             }
         }
 
+
+        public async Task<List<LocalityCandidateInfo>> GetLocalityCandidates(int countryId, int skip, int top, CancellationToken cancellationToken)
+            => await _dbContext.LocalityCandidates
+                .OrderBy(lc => lc.Id)
+                .Skip(skip)
+                .Take(top)
+                .Where(lc => lc.CountryId == countryId)
```
Comment (hasmikharutyunyan) [inline HappyTravel.Nakijin.Api/Services/AdministratorServices/LocalityCandidatesManagementService.cs:134]: Where condition needed before Order, Skip, Take

File: HappyTravel.Nakijin.Api/Controllers/LocalitiesManagementController.cs [hunk 2]
```diff
@@ -54,7 +56,27 @@ public async Task<IActionResult> MoveLocalityToSynonyms([Required] [FromRoute] s
...
+
+        /// <summary>
+        /// Retrieves locality synonyms
+        /// </summary>
+        /// <param name="localityHtId"></param>
+        /// <param name="cancellationToken"></param>
+        /// <returns></returns>
+        [ProducesResponseType(typeof(List<LocalitySynonymInfo>), StatusCodes.Status200OK)]
+        [ProducesResponseType(typeof(ProblemDetails), StatusCodes.Status400BadRequest)]
+        [HttpGet("localities/{localityHtId}/locality-synonyms")]
```
Comment (hasmikharutyunyan) [inline HappyTravel.Nakijin.Api/Controllers/LocalitiesManagementController.cs:69]: `localities/{localityHtId}/synonyms`  ?

File: HappyTravel.Nakijin.Api/Models/LocalityCandidates/LocalityCandidateInfo.cs [hunk 3]
```diff
@@ -0,0 +1,19 @@
+﻿using HappyTravel.MultiLanguage;
+
+namespace HappyTravel.Nakijin.Api.Models.LocalityCandidates
+{
+    public readonly struct LocalityCandidateInfo
```
Comment (hasmikharutyunyan) [inline HappyTravel.Nakijin.Api/Models/LocalityCandidates/LocalityCandidateInfo.cs:5]: maybe needed supplier info here?

File: HappyTravel.Nakijin.Api/Models/LocalityCandidates/LocalitySynonymInfo.cs [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+            ExLocalityId = exLocalityId;
+            SynonymType = synonymType;
+        }
+
+
+        public int Id { get; init; }
+        public MultiLanguage<string> Names { get; init; }
+        public int LocalityId { get; init; }
+        public int? ExLocalityId { get; init; }
+        public LocalitySynonymTypes SynonymType { get; init; }
```
Comment (hasmikharutyunyan) [inline HappyTravel.Nakijin.Api/Models/LocalityCandidates/LocalitySynonymInfo.cs:22]: maybe by  `ExLocalityId` it is nullable or not we can understand synonymType ? Or You think in the future will be another types?

File: HappyTravel.Nakijin.Api/Controllers/LocalitiesManagementController.cs (hunk 2 above)
Comment (oleg-n-k) [reply HappyTravel.Nakijin.Api/Controllers/LocalitiesManagementController.cs:69]: fixed

File: HappyTravel.Nakijin.Api/Models/LocalityCandidates/LocalityCandidateInfo.cs (hunk 3 above)
Comment (oleg-n-k) [reply HappyTravel.Nakijin.Api/Models/LocalityCandidates/LocalityCandidateInfo.cs:5]: supplier info isn't stored in locality candidates table

File: HappyTravel.Nakijin.Api/Models/LocalityCandidates/LocalitySynonymInfo.cs (hunk 4 above)
Comment (oleg-n-k) [reply HappyTravel.Nakijin.Api/Models/LocalityCandidates/LocalitySynonymInfo.cs:22]: Yes, there will be other types

**Review by hasmikharutyunyan — APPROVED** [review]

**Review by AlexeyShok — APPROVED** [review]

**Pull request merged**